common/console.cpp

```diff
@@ -111,7 +111,16 @@ namespace console {
             }
         }
 
-        setlocale(LC_ALL, "");
+        auto locale = setlocale(LC_ALL, "");
+        auto lang = getenv("LANG");
+
+        if (locale == nullptr) {
```

Comment:
This seems like a reasonable fix to me. Since it's not often needed, it may be better to defer the retrieval of the "LANG" environment variable until it's needed, but it's good as is.

```suggestion
        if (locale == nullptr) {
            auto lang = getenv("LANG");
```

Reply:
I found one more case, default debian 12 minimal image, and probably more default images for vm/lxc/docker can have a non utf `LC_ALL` and non utf `lang`, so just one more commit.